include/highfive/H5DataType.hpp

```diff
@@ -158,6 +158,67 @@ class CompoundType : public DataType {
     void create(size_t size = 0);
 };
 
+///
+/// \brief Create a enum HDF5 datatype
+///
+/// \code{.cpp}
+/// enum class Position {
+///     FIRST = 1,
+///     SECOND = 2,
+/// };
+///
+/// EnumType<Position> create_enum_position() {
+///     return EnumType<Position>({{"FIRST", Position::FIRST},
+///                                {"SECOND", Position::SECOND}});
+/// }
+///
+/// template<>
+/// DataType create_datatype<Position>() {
+///     return create_enum_position();
+/// }
+///
+/// void write_first(H5::File& file) {
+///     auto en = create_enum_position();
+///     en.commit(file, "Position");
+///     auto dataset = file.createDataset("/foo", DataSpace(1), en);
+///     dataset.write(Position::FIRST);
+/// }
+/// \endcode
+template<typename T>
+class EnumType: public DataType {
+public:
+    ///
+    /// \brief Use for defining a member of enum type
+    struct member_def {
+        member_def(std::string t_name, T t_value)
+            : name(std::move(t_name))
+            , value(std::move(t_value)) {}
+        std::string name;
+        T value;
+    };
+
+    EnumType(const EnumType& other) = default;
+
+    EnumType(std::vector<member_def> t_members)
+        : members(std::move(t_members)) {
+        create();
+    }
+
+    template<typename U>
+    EnumType(const std::initializer_list<member_def>& t_members)
```

Comment:
I've recently read that initialiser lists (and btw, also iterators and function objs) are better passed by value: https://stackoverflow.com/questions/17803475/why-is-stdinitializer-list-often-passed-by-value. I think the difference is negligible though